src/main/CommandLine.cpp

```diff
@@ -1960,6 +1936,38 @@ runApplyLoad(CommandLineArgs const& args)
 
                 ApplyLoad al(app, mode);
 
+                // In the limit-based mode, we may publish the history
+                // checkpoint just before performing the benchmark. This way
+                // the checkpoint could be used for both setting up the
+                // throughput benchmark downstream, and also potentially for
+                // speeding up the subsequent benchmark runs via catchup (that's
+                // not supported yet though).
+                if (mode == ApplyLoadMode::LIMIT_BASED &&
+                    app.getHistoryArchiveManager().publishEnabled())
+                {
+                    app.getHistoryManager().waitForCheckpointPublish();
```

Comment:
Minor: the flow would be more intuitive if we always called closeLedger (which would maybe enqueue the next checkpoint), then always call `waitForCheckpointPublish` which would always ensure queue size is 0. Manually calling `maybeQueueHistoryCheckpoint` seems a bit suspicious, since closeLedger also calls the same method.